tests/modules/test_watershed.py

```diff
@@ -0,0 +1,71 @@
+import cellprofiler.modules.watershed
+import cellprofiler.image
+import numpy
+import numpy.testing
+import skimage.filters.rank
+import skimage.measure
+import skimage.morphology
+
+instance = cellprofiler.modules.watershed.Watershed()
+
+
+def test_run(image, module, image_set, workspace):
+    module.x_name.value = "gradient"
+
+    module.y_name.value = "watershed"
+
+    module.markers_name.value = "markers"
+
+    if image.multichannel or image.dimensions is 3:
+        denoised = numpy.zeros_like(image.pixel_data)
+
+        for idx, data in enumerate(image.pixel_data):
+            denoised[idx] = skimage.filters.rank.median(data, skimage.morphology.disk(2))
+    else:
+        denoised = skimage.filters.rank.median(image.pixel_data, skimage.morphology.disk(2))
```

Comment:
We do this pattern often. Until `skimage.filters.rank` supports more than single-channel images, can we create a wrapper function? E.g.,

```python
def apply(filter, image, selem):
    if image.ndim is 2:
        return getattr(skimage.filters.rank, operation)(image.pixel_data, selem)
    
    result = numpy.zeros_like(image.pixel_data)

    for idx, data in enumerate(image.pixel_data):
        result[idx] = getattr(skimage.filters.rank, operation)(data, selem)

    return result
```